src/rules/at-rule-empty-line-before/index.js

```diff
@@ -74,5 +75,17 @@ export default function (expectation, options) {
         )
       }
     })
+
+    validateOptions({ result, ruleName,
```

Comment:
This might seem like an odd request, but is there any chance of getting this placed at the beginning of the export function? e.g.

```js
/**
 * @param {"always"|"never"} expectation
 * @param {object} options - Can contain the following:
 *   - except: ["blockless-group", "first-nested", "all-nested"]
 *   - ignore: ["all-nested"]
 */
export default function (expectation, options) {
  return (root, result) => {

    validateOptions({ result, ruleName,
      actual: expectation,
      possible: [ "always", "never" ],
    })
    validateOptions({ result, ruleName,
      actual: options,
      possible: {
        except: [ "blockless-group", "first-nested", "all-nested" ],
        ignore: ["all-nested"],
      },
    })

    root.eachAtRule(atRule => { ... })
  }
}
```

I'd prefer it because:

1. It's closer to the jsdoc, so I think it'll be easier to spot disparities between them.
2. I think it feels like it flows a little better e.g. "right, lets check the options first... and then process".
3. I think having it at the top would encourage plugin developers to use the validation utils. I can see it being forgotten about if we leave it at the bottom.